### DISCUSSION TITLE
How to ensure script follows immediately after <body> tag and not before </body> tag?

### DISCUSSION THREAD

**Discussion by glennlea1525** [squidfunk/mkdocs-material#6411] [General]
I used a suggestion found here [https://github.com/squidfunk/mkdocs-material/discussions/5607#discussioncomment-7323367](https://github.com/squidfunk/mkdocs-material/discussions/5607#discussioncomment-7323367) to add a Google Tag Manager script to my Materials project. It worked, except the override in main.html placed the script just before the closing body tag. However, it should be placed immediately after the opening body tag. I don't know how to do this without making a mess of the project. 

The GTM instructions are here: [Google Tag Manager Support](https://support.google.com/tagassistant/answer/3207128?hl=en#script_flow)

Thanks for any assistance or suggestions.

**Comment by kamilkrzyskow** [comment]
Hello,
while it says it should follow the `body` tag immediately, it also says it shouldn't be inside any other HTML tag, so that second condition is met and I think it shouldn't cause any issues. Especially if you add a defer to that script tag.

As for an exact solution you'd have to override the `base.html` template directly, since there is no convenient block near the start of the body tag.

Overriding the `base.html` file is discouraged as it would force you to basically maintain the file after each update to the template.

However, you could attempt to override it dynamically during docs build:
https://github.com/squidfunk/mkdocs-material/discussions/5060
The guide is slightly outdated, as I've switched to using two separate hooks and might later turn it into a plugin. You can get newer files here:
https://github.com/kamilkrzyskow/gmc/tree/upgrade-material-9.4/overrides/.hooks
The current idea is that you hook into the theme_overrides_manager hook, and you can see how the language_flags hook does it.
